README.md

```diff
@@ -55,13 +64,130 @@ You can change the generated files in `InitCommand` and your example project in
+                                             push: false))
+```
+
+Let's see how you can edit this by yourself to get the result you want.
+
+## TapestryConfig
+
+This currently only has one argument `relase`, more will be added here
+
+## Release
+
+The main component of the release process.
+
+| Property        | Description           | Type  | Optional      |   Default   |
+| ------------- |:-------------:| :-----:| :-----:| :-----:|
+| actions      | Your custom release steps, more about how you can define them below (see `ReleaseAction`) | [ReleaseAction] | No | []
+| add      | Glob of your files that you want to add to git before releasing a version   |   [String] |   Yes
+| commitMessage |  Commit message - use `Argument.version` in your string which will translate to the version you are releasing   |  String |  No
+| push      | This option determines if the changes are committed at the end | Bool | No | 
+
+Running `tapestry release 0.0.1` also automatically tags the latest commit.
+
+## ReleaseAction
+
+This type lets you define the individual actions. 
+
+| Case        | Description           | 
+| ------------- |:-------------:|
+| `pre(tool: String, arguments: [String] = [])`      | Runs before commiting and pushing. Runs the custom command from the root folder.
+| `pre(_ predefinedAction: PredefinedAction)`      | Runs before commiting and pushing. Runs a `PredefinedAction`, more on that below.
+| `post(tool: String, arguments: [String] = [])` |  Runs after commiting and pushing. Runs the custom command from the root folder.
+| `post(_ predefinedAction: PredefinedAction)`       | Runs after commiting and pushing. Runs a `PredefinedAction`, more on that below.
+
+
+## PredefinedAction
+
+This is a set of predefined actions.
+
+| Case        | Description           | 
+| ------------- |:-------------:|
+| `docsUpdate`      | Updates version in your `.podspec, README.md` and adds it in `CHANGELOG.md`
+| `run(tool: String, arguments: [String])`     | You can run defined dev dependencies in `tapestries/Package.swift`
+|  `dependenciesCompatibility([DependenciesManager])` |  Checks compatibility with given `DependenciesManager`s
+
+Note that for `docsUpdate` your changelog must be in this format:
+```
+## Next
+
+- your way of adding logs
+
+## 1.0.0
+....
+```
+
+Which will be changed to
+```
+## Next
+
+## 1.0.1
+
+- your way of adding logs
+
+## 1.0.0
+....
+```
+
+To change the version in your `README.md`, we use the latest tag in your git history.
+
+You can also try out the individual actions:
+```bash
+tapestry run action docs-update 1.0.0
+tapestry run action compatibilty spm carthage cocoapods
+```
+
+## DependenciesManager
+
+Let's you say what dependency managers you want to check compatibility for.
+
+| Case        | Description           | 
+| ------------- |:-------------:|
+| `cocoapods`      | Runs compatibility check for [Cocoapods](https://github.com/CocoaPods/CocoaPods)
```

Comment:
❕  maybe when using more links to the same page in the same markdown it would be better to use the "reference" syntax for links
`[cocoapods][Cocoapods] ` and at the end of document the list of references `cocoapods: https://github.com/CocoaPods/CocoaPods`

Reply:
Huh, did not know about this, thanks!
BTW, you have to use `[]` for the reference: 
```
[cocoapods]: https://github.com/CocoaPods/CocoaPods
```
Added.